src/components/ClassComponent/index.jsx

```diff
@@ -1,7 +1,89 @@
 import React, { Component } from "react";
+import "./index.css";
+const apiUrl = "https://official-joke-api.appspot.com/jokes/random";
+class ClassComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      setup: "",
+      punchline: "",
+      loading: false,
+    };
+    this.handleFetch = this.handleFetch.bind(this);
+  }
+
+  componentDidMount() {
+    console.log("component mounted");
+  }
+
+  componentDidUpdate() {
+    console.log("component updated");
+  }
+
+  componentWillUnmount() {
+    console.log("component unmounted");
+  }
+
+  handleFetch() {
+    this.setState({ loading: true });
+    fetch(apiUrl, {
+      method: "GET",
+    })
```

Comment:
No get n precisa definir de forma explícita, ele já assume por default se não tiver nada definido que é get
```suggestion
    fetch(apiUrl)
```